Convert swift4

JamesDurham opened the pull request:
A version of Realm Model Generator in swift 4.

JamesDurham committed:
Bumped the swift version number

JamesDurham committed:
convert the last of the AnyObject to Any

JamesDurham committed:
transform NibDesignable bases and extensions.

JamesDurham committed:
Fix what I could about NSPasteboard: EntitiesView.swift

JamesDurham committed:
Add recommended build settings for xcode 9

JamesDurham committed:
Revert "transform NibDesignable bases and extensions."

This reverts commit 6f657de2d8d0aaaafd47b99a91eae12aae1159a3.

JamesDurham committed:
NibDesignable change is back

JamesDurham committed:
view/base done

JamesDurham committed:
AttributesView done
make constants correct type
Paste Board stuff see:
"Fix what I could about NSPasteboard: EntitiesView.swift"

JamesDurham committed:
move constant types EntitiesView.swift

JamesDurham committed:
RelationshipsView.swift
make constants correct type
Paste Board stuff see:
"Fix what I could about NSPasteboard: EntitiesView.swift"

JamesDurham committed:
DetailsMainVC.swift done
make constants

JamesDurham committed:
EntityDetailView.swift done

JamesDurham committed:
AttributeDetailView.swift done

JamesDurham committed:
RelationshipDetailView.swift done
using identifier:NSUserInterfaceItemIdentifier
of identifier:String
in protocol

JamesDurham committed:
RealmSchemaDocument.swift done
extract constants

JamesDurham committed:
MainVCExtension.swift done 
move constant
file open panel callback change

berbschloe commented on RealmModelGenerator/AttributeDetailView.swift:
```diff
@@ -45,28 +46,28 @@ class AttributeDetailView: NibDesignableView, NSTextFieldDelegate {
-        set { self.indexedCheckBox.state = newValue ? 1 : 0 }
-        get { return self.indexedCheckBox.state == 1 ? true : false }
+        set { self.indexedCheckBox.state = newValue ? NSControl.StateValue.on : NSControl.StateValue.off }
+        get { return self.indexedCheckBox.state == NSControl.StateValue.on ? true : false }
```
use shorthand `.on` and `.off` where applicable.

JamesDurham replied:
Done

berbschloe commented on RealmModelGenerator/MainVCExtension.swift:
```diff
@@ -8,7 +8,11 @@
 import Cocoa
 
+// --MARK-- temporary workaround
+fileprivate let VERSIONVC: NSStoryboard.SceneIdentifier = NSStoryboard.SceneIdentifier(rawValue: "VersionVC")
```
static `let` are valid inside of extensions.
You could also declare at usage site. keeps the code "localized"

JamesDurham replied:
SceneIdentifier should be in seprate file (and other ids) 
just not now

berbschloe commented on RealmModelGenerator/NibDesignable.swift:
```diff
@@ -72,7 +72,7 @@ extension NSView {
     }
     
-    public func nibDidLoad() {}
+    // public func nibDidLoad() {}
```
remove commented out method. It's added below

JamesDurham replied:
Done

berbschloe reviewed (changes requested):
Just a couple nits. Other than that. Looks really good.

JamesDurham committed:
Fix enum reference style for NSControl.StateValue

JamesDurham committed:
Spelling

Extra (Not spelling):
RealmModelGenerator/RelationshipDetailView.swift
line 39, 40, 80, and 83 
NSControl.StateValue access transform

JamesDurham committed:
add discardableResult

JamesDurham committed:
give the test some love

JamesDurham committed:
Fix enum reference style for AttributeType

JamesDurham committed:
Fix enum reference style for Language

JamesDurham committed:
VERSIONVC static constant converging to a standard.

JamesDurham committed:
remove commented code.
